activerecord/lib/active_record/locking/pessimistic.rb

```diff
@@ -59,7 +59,15 @@ module Pessimistic
       # or pass true for "FOR UPDATE" (the default, an exclusive row lock). Returns
       # the locked record.
       def lock!(lock = true)
-        reload(:lock => lock) if persisted?
+        if persisted?
+          if changed?
+            ActiveSupport::Deprecation.warn(<<-MSG.squish)
+              WARNING! Locking a record with unpersisted changes is deprecated. Use
```

Comment:
`Locking a record with unpersisted changes is deprecated and will raise an exception in Rails 5.1. Use`